server/shared/src/main/scala/org/http4s/server/middleware/DefaultHead.scala

```diff
@@ -35,17 +35,23 @@ import org.http4s.Method.HEAD
   * requiring more optimization should implement their own HEAD handler.
   */
 object DefaultHead {
-  def apply[F[_]: Functor, G[_]: Concurrent](http: Http[F, G])(implicit F: MonoidK[F]): Http[F, G] =
+  def apply[F[_]: Functor, G[_]: Applicative](
+      http: Http[F, G]
+  )(implicit F: MonoidK[F]): Http[F, G] =
     Kleisli { req =>
       req.method match {
         case HEAD => http(req) <+> http(req.withMethod(GET)).map(drainBody[G])
         case _ => http(req)
       }
     }
 
-  def httpRoutes[F[_]: Concurrent](httpRoutes: HttpRoutes[F]): HttpRoutes[F] =
+  def httpRoutes[F[_]: Monad](httpRoutes: HttpRoutes[F]): HttpRoutes[F] =
```

Comment:
Just because it is source-compatible, it is not binary-compatible. Source-compatible changes are only effective if the entire ecosystem recompiles and publishes. To guarantee compatibility without forcing recompilation, we must preserve binary-compatibility.

Anyway it's easy to keep binary-compatibility here. Just add a deprecated signature:
```scala
def httpRoutes[F[_]](httpRoutes: HttpRoutes[F], F: Concurrent[F]): HttpRoutes[F]
```

Reply:
Alright, I did my best to preserve binary compatibility. Not sure it's a thing this time, but I'm not against it.

Reply:
> Not sure it's a thing this time, but I'm not against it.

Genuine question: why wouldn't it be a thing? Why is this situation different from all the other ones where we maintain binary-compatibility?

Reply:
If source compatibility isn't being hurt, I like refactorings. Probably, I mostly think from the user's perspective.

Reply:
For apps, source compatibility is more important: they are typically recompiled whenever http4s is upgraded.  For all the libraries that sit between http4s and an app, including our satellite repos, it's critical that we maintain binary compatibility, or else apps risk failing fatally at runtime.  It's not optional for 0.23, and why we take a very long time between breaking releases and batch them up.

Reply:
Doubtless, I was not arguing about preferring source compatibility over binary compatibility generally, only for this particular case of relaxing constraints.